libs/community/langchain_community/llms/oci_data_science_model_deployment_endpoint.py

```diff
@@ -82,36 +417,99 @@ def _identifying_params(self) -> Dict[str, Any]:
+    ) -> LLMResult:
+        """Call out to OCI Data Science Model Deployment endpoint with k unique prompts.
 
-    def _invocation_params(self, stop: Optional[List[str]], **kwargs: Any) -> dict:
-        """Combines the invocation parameters with default parameters."""
-        params = self._default_params
-        if self.stop is not None and stop is not None:
-            raise ValueError("`stop` found in both the input and default params.")
-        elif self.stop is not None:
-            params["stop"] = self.stop
-        elif stop is not None:
-            params["stop"] = stop
-        else:
-            # Don't set "stop" in param as None. It should be a list.
-            params["stop"] = []
+        Args:
+            prompts: The prompts to pass into the service.
+            stop: Optional list of stop words to use when generating.
 
-        return {**params, **kwargs}
+        Returns:
+            The full LLM output.
 
-    def _process_response(self, response_json: dict) -> str:
-        raise NotImplementedError
+        Example:
+            .. code-block:: python
+
+                response = oci_md.invoke("Tell me a joke.")
+                response = oci_md.generate(["Tell me a joke."])
+        """
+        generations: List[List[Generation]] = []
+        params = self._invocation_params(stop, **kwargs)
+        for prompt in prompts:
+            body = self._construct_json_body(prompt, params)
+            if self.streaming:
+                generation = GenerationChunk(text="")
+                for chunk in self._stream(
+                    prompt, stop=stop, run_manager=run_manager, **kwargs
+                ):
+                    generation += chunk
+                generations.append([generation])
+            else:
+                res = self.completion_with_retry(
+                    data=body,
+                    run_manager=run_manager,
+                    **kwargs,
+                )
+                generations.append(self._process_response(res.json()))
+        return LLMResult(generations=generations)
+
+    async def _agenerate(
+        self,
+        prompts: List[str],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> LLMResult:
+        """Call out to OCI Data Science Model Deployment endpoint async with k unique prompts.
+
+        Args:
+            prompts: The prompts to pass into the service.
+            stop: Optional list of stop words to use when generating.
+
+        Returns:
+            The full LLM output.
+
+        Example:
+            .. code-block:: python
 
-    def _call(
+                response = await oci_md.ainvoke("Tell me a joke.")
+                response = await oci_md.agenerate(["Tell me a joke."])
+        """  # noqa: E501
+        generations: List[List[Generation]] = []
+        params = self._invocation_params(stop, **kwargs)
+        for prompt in prompts:
```

Comment:
Maybe it is better to check if `prompts` is a list here so that user will not pass in a single string by mistake (which will cause the agenerate to run for a long time?
```
prompts = [prompts] if isinstance(prompts, str) else prompts
```

Reply:
added .